paimon-common/src/main/java/org/apache/paimon/utils/ThreadPoolUtils.java

```diff
@@ -120,17 +120,29 @@ private void advanceIfNeeded() {
     public static <U> void randomlyOnlyExecute(
             ExecutorService executor, Consumer<U> processor, Collection<U> input) {
         List<Future<?>> futures = new ArrayList<>(input.size());
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
         for (U u : input) {
-            futures.add(executor.submit(() -> processor.accept(u)));
+            futures.add(
+                    executor.submit(
+                            () -> {
+                                Thread.currentThread().setContextClassLoader(cl);
+                                processor.accept(u);
```

Comment:
After processor.accept(u); set the classloader back to previous?

Reply:
No need, the thread pool is only for this usage.